test/overlapped-checker/main.c

```diff
@@ -0,0 +1,23 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "../../deps/uv/include/uv.h"
+#include "../../deps/uv/src/uv-common.h"
+
+int main() {
+  uv_loop_t *loop = uv_default_loop();
+  uv_pipe_t pipe;
+
+  for (int i = 0; i <= 2; i++) {
+    if (uv_pipe_init(loop, &pipe, 0) || uv_pipe_open(&pipe, i)) {
+      abort();
+    }
+
+    int non_overlapped_flag = !!(pipe.flags & UV_HANDLE_NON_OVERLAPPED_PIPE);
+
+    fprintf(stderr, "%d", non_overlapped_flag);
```

Comment:
You should fflush() or print a newline on exit because stderr might be in line-buffered mode, see `man stdbuf`.